### ISSUE TITLE
Support clean URLs and dropping trailing slashes from the pub docs server

### ISSUE THREAD

**Issue by chalin** [dart-lang/pub-dartlang-dart#1218]
Site-webdev is configured in this way. It makes use of the following firebase config parameters: [clean URLs](https://firebase.google.com/docs/hosting/full-config#cleanurls) and  [(dropping) trailing slashes](https://firebase.google.com/docs/hosting/full-config#trailingslash).

cc @kwalrath @kevmoo

**Comment by isoos** [comment]
@chalin: do you mean using 
- `https://pub.dartlang.org/documentation/pana/0.10.6` instead of
- `https://pub.dartlang.org/documentation/pana/0.10.6/` ?

Unfortunately the current dartdoc template uses `base` reference to link to files, and that requires us to use the trailing slash. Even if we output the URL without trailing slash, the server needs to redirect it to use the trailing one.

**Comment by chalin** [comment]
Ok, how about the "clean URLs" feature?

**Comment by isoos** [comment]
It would be a nice thing to have, but it would require changes in how dartdoc generates its files and its links, or the pub site would need to do much more post-processing there. I don't think it is a priority right now.

**Comment by chalin** [comment]
Just to clarify: it should just be a server config parameter. There would be nothing to change in the way links are generated (links inside docs can still have .html, but links wouldn't be _required_ to end in .html). I understand that this might not be a priority.

**Comment by kevmoo** [comment]
We could open an associated issue on the dartdoc repo – CC @jcollins-g

**Comment by isoos** [comment]
>  it should just be a server config parameter

Is firebase hosting the same as appengine bucket? Because we are using the later...

Also: we are not exposing the bucket directly, because we are storing the content under an uuid-based directory. This allows many features (e.g. bad upload protection, controlled version upgrades).

**Comment by sigurdm** [comment]
I don't think we are going to get around to this in any foreseeable future. And having `.html` doesn't seem too harmful.

Closing as Wontfix.

**Comment by kevmoo** [comment]
CC @srawlins – we'd have to do the work in dartdoc first, obviously. If/when that happens it's just a matter of supporting `index.html` correct on the pub infra, which I'm guessing will be trivial ask.
